
# rust-1.39 deprecated the "try!" call in favor of "?" which then breaks
# nsstring because it treats warnings as errors.

diff -Nrup -U 8 a/mozilla/xpcom/rust/nsstring/src/lib.rs b/mozilla/xpcom/rust/nsstring/src/lib.rs
--- a/mozilla/xpcom/rust/nsstring/src/lib.rs	2020-02-25 03:10:53.802918476 +0300
+++ b/mozilla/xpcom/rust/nsstring/src/lib.rs	2020-02-25 00:30:21.593463335 +0300
@@ -108,17 +108,16 @@
 //! similar to an `ns[C]String<'static>`, however, it does not have a `Drop`
 //! implementation.
 //!
 //! If this type is dropped in rust, it will not free its backing storage. This
 //! can be useful when implementing FFI types which contain `ns[C]String` members
 //! which invoke their member's destructors through C++ code.
 
 #![allow(non_camel_case_types)]
-#![deny(warnings)]
 
 #[macro_use]
 extern crate bitflags;
 
 use std::ops::{Deref, DerefMut};
 use std::marker::PhantomData;
 use std::borrow;
 use std::slice;
